html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Login.pm

```diff
@@ -145,30 +146,49 @@ sub authenticate {
         return;
     }
 
-    if(isenabled($self->app->profile->reuseDot1xCredentials)) {
+    if ( isenabled($self->app->profile->reuseDot1xCredentials) ) {
         my $mac       = $self->current_mac;
         my $node_info = node_view($mac);
         ($username,$realm) = strip_username($node_info->{'last_dot1x_username'});
-        get_logger->info("Reusing 802.1x credentials. Gave username ; $username");
+        $realm = lc($pf::constants::realm::NULL) unless(defined($realm));
+        get_logger->info("Reusing 802.1x credentials with username '$username' and realm '$realm'");
+
+        # Fetch appropriate source to use with 'reuseDot1xCredentials' feature
+        my $source = pf::config::util::get_realm_source($username, $realm);
+        
+        # No source found for specified realm
+        unless ( defined($source) ) {
+            get_logger->error("Unable to find an authentication source for the specified realm '$realm' while using reuseDot1xCredentials");
+            $self->app->flash->{error} = "Cannot find a valid authentication source for '" . $node_info->{'last_dot1x_username'} . "'";
+
+            $self->done();
```

Comment:
Done actually marks this module as complete and continues the flow.

`done != detach`

Not sure that is really what you want